docs/deployment/debian_ubuntu.md

```diff
@@ -0,0 +1,236 @@
+
+Debian and Ubuntu related distros have their applications packaged in `.deb` files, that can be installed via `sudo apt install ./your_package.deb`.
+
+## Tutorial
+
+In this tutorial, we will use the `dpkg-deb` tool to compile your `.deb` package.
+
+### 1) Organize program files in a staging folder
+
+Debian packages follow this basic structure:
+
+```sh
+./staging_folder/
+├── DEBIAN
+│   └── control # package control file
+└── usr
+    ├── bin
+    │   ├── your_program # starter script (optional)
+    │   └── your_program_executable
+    ├── lib
+    │   └── your_program
+    │       ├── libHarfBuzzSharp.so # Avalonia native library
+    │       ├── libSkiaSharp.so # Avalonia native library
+    │       └── other_native_library_1.so
+    └── share
+        ├── applications
+        │   └── YourProgram.desktop # desktop shortcut file
+        ├── icons
+        │   └── hicolor
+        │       ├── ... # other resolution icons (optional)
+        └── pixmaps
+            └── your_program.png # main application icon
+```
+
+Meaning of each folder:
+
+* `DEBIAN`: contains the `control` file.
+* `/usr/bin/`: contains the main executable and its starting script (recommended for starting your program via command-line).
+* `/usr/lib/your_program/`: where the `.so` native libraries will be included. (`your_program` is custom)
+* `/usr/share/applications/`: where the desktop shortcut will be placed.
+* `/usr/share/pixmaps/` and `/usr/share/icons/hicolor/**`: folders for application icons.
+
+:::info
+
+The `/usr/share/icons/hicolor/**` are *optional*, as in your app icon will probably show up on desktop even without those images, however, it is recommended to have them for better resolution.
+
+:::
+
+### 2) Make the `control` file
+
+The `control` file goes inside the `DEBIAN` folder.
+
+This file describes general aspects of your program, such as its name, version, category, maintainer, processor architecture and licenses. [Debian docs](https://www.debian.org/doc/debian-policy/ch-controlfields.html) have a more thorough description of all possible fields in the file.
+
+:::tip
+
+Author's comment: don't worry too much about filling all possible fields, most aren't required. This tutorial is to make a "good-enough" Debian package.
+
+:::
+
+Below is a simple example of a `control` file.
+
+```
+Package: your_program
+Version: 3.1.0
+Section: devel
+Priority: optional
+Architecture: amd64
+Installed-Size: 68279
+Maintainer: Ken Lee <kenlee@outlook.com>
+Homepage: https://github.com/kenlee/your_program
+Description: This is YourProgram, great for doing X.
+Copyright: 2022-2024 Ken Lee <kenlee@outlook.com>
```

Comment:
Add info about dependencies. 

.NET Runtime requires at least: `ca-certificates,  libc6,  libstdc++6, tzdata, zlibg1`, some version of `libicu` (an entry that looks like `libicu60 | libicu61 | libicu62 | ...` would suffice), some version of `libssl` (`libssl1.1 | libssl3.0`), `libgcc_s.so.1` (distro-version specific, can be `libgcc1 | libgcc-s1`

Avalonia requires at least `libx11-6, libice6, libsm6, libfontconfig1`